doc/VTGateV3Features.md

```diff
@@ -0,0 +1,86 @@
+
+#### Shared indexes
+There are situations where multiple tables share the same foreign key. A typical use case is a situation where there is a customer table, an order table and an order_detail table. The order table would have a customer_id column. In order to efficiently access all orders of a customer, it would be beneficial to shard this table by customer_id. This will co-locate order rows with their corresponding customer row.
+
+The order table would also need an order_id column. As mentioned above, you can create an order_id->customer_id cross-shard index for this table. This will allow you to efficiently access orders by their order_id.
+
+In the case of an order_detail table, it may only need an order_id foreign key. Since this foreign key means the same thing as the order_id in order, creating a cross-shard index for it will result in a duplication of the order_id->customer_id index. In such situations, V3 allows you to just reuse the existing index for the order_detail table also. This saves disk space and also reduces the overall write load.
+
+### Consistency
+Once you add multiple indexes to tables, it's possible that the application could make inconsistent requests. V3 makes sure that none of the specified constraints are broken. For example, if a table had both a basic sharding key and a hashed sharding key, it will enforce the rule that the hash of the basic sharding key matches that of the hashed sharding key.
+
+Some of the changes require updates to be performed across multiple databases. For example, inserting a row into a table that has a cross-shard key requires an additional row to be inserted into the lookup table. This results in distributed transactions. Currently, this is a best effort update. It is possible that partial commits happen if databases fail in the middle of a distributed commit.
+
+*There is a plan to support 2PC transactions to overcome this limitation.*
+
+### Query diversity
+V3 does not support the full SQL feature set. The current implementation supports simple queries:
+* Single table DML statements: This is a vitess-wide restriction where you can affect only one table and one sharding key per statement. *This restriction may be removed in the future.*
+* Single table SELECT statements:
+  * All constructs allowed if the statement targets only a single sharding key
+  * Aggregation and sorting not allowed if the statement targets more than one sharding key. Selects are allowed to target multiple sharding keys as long as the results from individual shards can be simply combined together to form the final result.
+
+Work is underway to support the following additional constructs:
+* Joins that can be served by sending the query to a single shard.
+* Joins that can be served by sending the query to multiple shards, and trivially combined to form the final result.
+* Cross-shard joins that can be served through a simple nested lookup.
```

Comment:
What's the difference between this and the previous item? I don't understand it. Maybe reformulate?